src/agent/state.py

```
@@ -129,6 +168,58 @@ def _merge_hvac(old: HVACSchema, new: HVACSchema) -> HVACSchema:
+    if idf is None:
+        return False
+    try:
+        return any(idf.all_of_type(t) for t in (
+            "Zone", "Material", "Material:NoMass", "Material:AirGap",
+            "WindowMaterial:SimpleGlazingSystem", "Construction",
+            "BuildingSurface:Detailed", "FenestrationSurface:Detailed",
+            "Schedule:Compact", "ScheduleTypeLimits",
+            "HVACTemplate:Thermostat", "HVACTemplate:Zone:IdealLoadsAirSystem",
+            "People", "Lights",
+        ))
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)